proposals/0173-Read-Generic-Network-Signal-data.md

```diff
@@ -16,277 +16,306 @@ To enable OEM proprietary mobile apps to read raw CAN/Network data by specifying
 
 ## Proposed solution
 
-Solution is to provide a Generic RPC, which can GET/SUBSCRIBE/UNSUBSCRIBE to any available Vehicle Data. 
-* Mobile app would pass a unique identifier and optionally an array of attributes through this RPC. These attributes are required to identify vehicle data requested by the app.
-* These attributes are proprietary for OEMs, so Mobile app needs to have access to proprietary interface definitions for vehicle data/service layer with which SDL interfaces to access vehicle data.
-* OEM implementation needs to enforce tracking of “unique identifier” to avoid duplicate subscriptions and to optimize subscription requests from multiple apps.
-* Mobile App is responsible to keep track of vehicle data it requests with help of field called "Id".
-* Mobile App can pick from three modes:
-  * GET		: Used for one time request to read current state/value of vehicle data
-  * SUBSCRIBE	: Used to receive updates to vehicle data for the period of subscription
-  * UNSUBSCRIBE: Used to stop the updates to vehicle data which was subscribed earlier
+Solution is to replace the way vehicle data items are validated for _GetVehicleData/SubscribeVehicleData/UnsubscribeVehicleData/OnVehicleData_ APIs. 
 
-**Request XML:**
+Instead of validating vehicle data items against APIs XML, SDL would rely on a dedicated vehicle data items schema as below:
 ```
-<function name="GenericNetworkData" functionID="GenericNetworkDataID" messagetype="request">
-    <description>Vehicle network data request</description>
+{
+	"id":"VehicleDataItem",
+	"dataType":"SDLDataType",
+	"reference":"OEMDataRef",
+	"array":true/false,
+	"mandatory":true/false
+}
+```
+* VehicleDataItem - It is the vehicle data item in question. e.g. gps, speed etc. SDL would use this as vehicle data param for requests from App and to validate policies permissions
+* SDLDataType - Return data type of the vehicle data item. It can be generic SDL data type e.g. Integer, String, Float etc OR an Enumeration defined in APIs XML. For vehicle data item which has sub-params, this would be Struct.
```

Comment:
Change to:
SDLDataType - **is the** return data type of the vehicle data item. It can be **a** generic SDL data type e.g. **integer**, **string**, **float**, **etc.** OR an **enumeration** defined in APIs XML. For **the** vehicle data item **that has** sub-params, this would be **struct**.